subprojects/instant-execution/src/integTest/groovy/org/gradle/instantexecution/InstantExecutionBuildScanIntegrationTest.groovy

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.instantexecution
+
+import org.gradle.util.Requires
+import org.gradle.util.TestPrecondition
+import spock.lang.Unroll
+
+
+class InstantExecutionBuildScanIntegrationTest extends AbstractInstantExecutionIntegrationTest {
+
+    def "--scan -Dscan.dump works"() {
+        given:
+        settingsKotlinFile << '''
+            plugins {
+                `gradle-enterprise`
+            }
+
+            gradleEnterprise.buildScan {
+                termsOfServiceUrl = "https://gradle.com/terms-of-service"
+                termsOfServiceAgree = "yes"
+            }
+        '''
+
+        when:
+        instantRun "help", "--scan", "-Dscan.dump"
+
+        then:
+        postBuildOutputContains("Build scan written to")
+
+        when:
+        instantRun "help", "--scan", "-Dscan.dump"
+
+        then:
+        postBuildOutputContains("Build scan written to")
+    }
+
+    @Requires(TestPrecondition.ONLINE)
```

Comment:
❓ any chance of testing this with a server OR mock local to the test?

Reply:
I did the same as in other places where publishing build scans is tested.
Just dumping the scan skips a lot of the build scan configuration.

Reply:
I don't mean dumping the scan but isn't there really any other way to prove build scan publishing is working other than connecting to a remote server?

Reply:
No fixture that I am aware of. I ended up removing that test given the one right below should already cover enough.